### DISCUSSION TITLE
Learning edge weights

### DISCUSSION THREAD

**Discussion by edgarsmdn** [pyg-team/pytorch_geometric#7804] [Q&A]
Hello!

I am wondering if the following task has been tackle already maybe in other contexts:

## Task
I have a graph for which I would like to **learn the edge weights**. In my dataset, the nodes and connectivity of the graph remain the same, but for different graph-level feature values I have different graph-level output values. From this I would like to learn the edge weights of the graph depending on the given graph-level feature value.

In general, I would be grateful if you can point me to examples of edge weight learning. Thank you!!

**Comment by wsad1** [comment]
Check out this [example](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/link_pred.py) which is a link prediction problem. You could simply change the problem to  a link regression problem ie. learn edge weights.
